### DISCUSSION TITLE
flink application模式下自动生成的启动命令是错误的

### DISCUSSION THREAD

**Discussion by huotianyu-CAa** [apache/dolphinscheduler#13688] [Q&A]
flink version:1.16.1
dolphinscheduler version:3.13

当我配置一个flink任务，并指定了jobmanager内存数，taskmananger内存数，slot数量和并行数量时，生成的启动命令如下：
flink run-application -t yarn-application -ys 4 -yjm 1024m -ytm 2048m -yqu default -p 4 -sae -c org.umpay.job.sql.Mysql2DorisFlinkSql flink_job/flink-job-1.0-SNAPSHOT-jar-with-dependencies-pt.jar

出现的问题就是，指定的参数在yarn集群申请容器时没有生效，只申请了一个jobmanager=1G,taskmanager=1G,solt=1的容器

通过搜索得知，application模式下指定yarn配置需要通过-D来指定，如下：
flink-1.16.1/bin/flink run-application -t yarn-application  -Djobmanager.memory.process.size=2048m -Dtaskmanager.memory.process.size=2048m -Dtaskmanager.numberOfTaskSlots=4 ./flink-job-1.0-SNAPSHOT-jar-with-dependencies-spt.jar 

请问是我的配置有问题，还是这确实是一个bug，该如何解决呢

**Comment by Radeity** [comment]
Hi, @huotianyu-CAa , if you find these configurations are unavailable for application mode, and have some relative doc to verify it, welcome to submit a PR to solve that problem, thx : )
